app/javascript/v3/helpers/OnboardingHelper.js

```diff
@@ -0,0 +1,16 @@
+export const findMatchingOption = (value, options, defaultValue) => {
+  const match = options.find(
+    option => option.value === value || option === value
+  );
+  return match ? match.value || match : defaultValue;
+};
+
+export const findCompanySizeMatch = (options, size) => {
+  if (!size || Number.isNaN(size)) return undefined;
+  return (
+    options.find(option => {
+      const upperLimit = option.value.split('-')[1]?.split('+')[0];
+      return size < (upperLimit ? Number(upperLimit) : Infinity);
+    })?.value || options[0].value
```

Comment:
I'm not sure I understand the code here. Why are returning Infinity? This function doesn't explain itself easily.

Reply:
I have added comments about this piece of code. We are parsing the upper limit and trying to see if it fits in that range. For the last option 1000+ there is no upper limit. We are using the upper limit is `Infinity `